unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Billboard/BillboardsController.cs

```diff
@@ -0,0 +1,92 @@
+using DCL;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BillboardsController : IBillboardsController
+{
+    private readonly List<Billboard> billboards = new List<Billboard>();
+    private Coroutine updateCoroutine;
+
+    private Vector3Variable CameraPosition => CommonScriptableObjects.cameraPosition;
+    private Vector3 lastCamPosition;
+    private bool camUpdated;
+
+
+    public static BillboardsController Create()
+    {
+        BillboardsController controller = new BillboardsController();
+
+        return controller;
+    }
+
+    public void Initialize()
+    {
+        updateCoroutine = CoroutineStarter.Start(UpdateCoroutine());
+    }
+
+    public void BillboardAdded(GameObject billboardContainer)
+    {
+        if (billboardContainer.TryGetComponent<Billboard>(out var billboard))
+        {
+            billboards.Add(billboard);
+            ChangeOrientation(billboard);
+        }
+    }
+
+    public void BillboardRemoved(GameObject billboardContainer)
+    {
+        Billboard billboard = billboardContainer.GetComponent<Billboard>();
+        if (billboard == null || !billboards.Contains(billboard))
+            return;
+
+        billboards.Remove(billboard);
+    }
+
+    public void Dispose()
+    {
+        CoroutineStarter.Stop(updateCoroutine);
+        billboards.Clear();
+    }
+
+
+    private void ChangeOrientation(Billboard billboard)
+    {
+        if (billboard.EntityTransform == null)
+            return;
+
+        Vector3 lookAtVector = billboard.GetLookAtVector(CameraPosition);
+        if (lookAtVector != Vector3.zero)
+            billboard.EntityTransform.forward = lookAtVector;
+    }
+
+
+    private IEnumerator UpdateCoroutine()
+    {
+        WaitForEndOfFrame waitForFrameEnd = new WaitForEndOfFrame();
+
+        while (true)
+        {
+            List<Billboard> currentBillboards = new List<Billboard>(billboards);
```

Comment:
Do we really need to copy the list every iteration?
If the list is changed while iterating on it we can avoid issues by using an `inverse-for`

Reply:
Considering the list is removed from and added to from Awake and OnDestroy, the list is not changed while iterating

Reply:
The first idea was to calculate on more than one frame (hence not allowing the list to be modified in the frame wait) if there where too many billboards, as the result was visible (meaning when a high number of billboards were present some of them updated weirdly), I removed the feature and now creating a new list is not needed anymore